src/Adapter/CMS/Page/QueryHandler/GetCmsPageForEditingHandler.php

```diff
@@ -66,8 +77,9 @@ public function handle(getCmsPageForEditing $query)
                 $cms->content,
                 $cms->indexation,
                 $cms->active,
-                $cms->getAssociatedShops()
-            );
+                $cms->getAssociatedShops(),
+                $preview_url = $this->context->link->getCMSLink($cms, null, null, $this->context->language->id)
```

Comment:
Why assigning `$preview_url` if you do not use it later ?

Reply:
I use its getter in cmsController editAction (To pass preview url to twig)

Reply:
I mean you could do
```suggestion
                $this->context->link->getCMSLink($cms, null, null, $this->context->language->id)
```

Why the `$preview_url =` part ? 😉

Reply:
oh for sure, i was missing the point. Its a cp mistake, you are right :D

Reply:
@matks fixed this